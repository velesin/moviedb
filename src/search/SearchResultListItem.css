.SearchResultListItem {
    margin-top: 64px;
    width: 300px;
    height: 450px;
    border-radius: 8px;
    overflow: hidden;
}
.SearchResultListItem__card {
    position: relative;
    width: 100%;
    height: 100%;
    transition: all 0.6s ease;
    transform-style: preserve-3d;
}
.SearchResultListItem:hover .SearchResultListItem__card {
    transform: rotateY(180deg);
}
.SearchResultListItem__front, .SearchResultListItem__back {
    background-color: #ffffff;
    box-sizing: border-box;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    backface-visibility: hidden;
}
.SearchResultListItem__back {
    padding: 16px;
    transform: rotateY(180deg);
    display: flex;
    flex-direction: column;
}
.SearchResultListItem__title {
    font-size: 24px;
}
.SearchResultListItem__synopsis {
    box-sizing: border-box;
    overflow: hidden;
    flex-grow: 1;
    padding: 16px 0 16px 0;
}
.SearchResultListItem__meta {
    display: flex;
}
.SearchResultListItem__date {
    flex-grow: 1;
}