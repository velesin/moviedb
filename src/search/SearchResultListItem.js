import React from 'react';
import { PosterImage } from '../shared/PosterImage';

export function SearchResultListItem({ movie }) {
    return (
        <div className="SearchResultListItem">
            <PosterImage path={ movie.poster_path} title={ movie.title } />
            <div className="SearchResultListItem__title">{ movie.title }</div>
            <div className="SearchResultListItem__date">{ movie.release_date }</div>
            <div className="SearchResultListItem__rating">{ movie.vote_average }</div>            
        </div>
    );
}