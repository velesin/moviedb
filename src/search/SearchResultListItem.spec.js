import React from 'react';
import { shallow } from 'enzyme';

import { SearchResultListItem } from './SearchResultListItem';

const MOVIE = {
    title: 'TITLE',
    release_date: '2010-03-12',
    vote_average: 5.4
};

it('renders correctly formatted movie list item', () => {
    const item = shallow(<SearchResultListItem movie={ MOVIE } />);
    expect(item.getElement()).toMatchSnapshot();
});
