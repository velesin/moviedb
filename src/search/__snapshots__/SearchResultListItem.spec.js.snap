// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly formatted movie list item 1`] = `
<div
  className="SearchResultListItem"
>
  <PosterImage
    path={undefined}
    title="TITLE"
  />
  <div
    className="SearchResultListItem__title"
  >
    TITLE
  </div>
  <div
    className="SearchResultListItem__date"
  >
    2010-03-12
  </div>
  <div
    className="SearchResultListItem__rating"
  >
    5.4
  </div>
</div>
`;
